spec/ship_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'rspec'
+require './lib/ship'
+
+RSpec.describe Ship do
+
+  # Test it can create new ships
+  it 'exits' do
+    cruiser = Ship.new("Cruiser", 3)
+
+    expect(cruiser).to be_instance_of(Ship)
+  end
+
+  # Test the ship has a name
+  it 'has a name' do
+    cruiser = Ship.new("Cruiser", 3)
+
+    expect(cruiser.name).to eq("Cruiser")
+  end
+
+  # Test the ship has length
+  it 'has length' do
+    cruiser = Ship.new("Cruiser", 3)
+
+    expect(cruiser.length).to eq(3)
+  end
+
```

Comment:
Not sure if this is overkill but to test dynamic functionality we could add tests with diff names and lengths.